packages/lesswrong/components/notifications/NotificationsPage/NotificationsPageKarmaChangeList.tsx

```diff
@@ -35,15 +35,40 @@ export const NotificationsPageKarmaChangeList = ({karmaChanges, truncateAt, clas
   let tagRevisions = karmaChanges?.tagRevisions
   
   // If we're truncating the list, attempt to only show the first n items.
-  // This doesn't quite work because some items can have more than one
-  // notification attached to them (like an upvote and an agree react).
   if (isTruncated && truncateAt !== undefined) {
-    let itemCount = truncateAt
-    posts = posts?.slice(0, itemCount)
-    itemCount -= posts?.length ?? 0
-    comments = comments?.slice(0, itemCount)
-    itemCount -= comments?.length ?? 0
-    tagRevisions = tagRevisions?.slice(0, itemCount)
+    let remainingItemCount = truncateAt
+    posts = posts?.slice(0, remainingItemCount)
+    // Count how many notifications the posts will show
+    const postNotificationCount = posts?.reduce((acc, post) => {
+      return acc + (post.scoreChange > 0 ? 1 : 0) + (Object.keys(post.eaAddedReacts ?? {}).length ?? 0)
```

Comment:
I think `(post.scoreChange > 0 ? 1 : 0)` (and the same for comments below) might need some check of `currentUser.karmaChangeNotifierSettings.showNegativeKarma`?

Reply:
Ah thanks for flagging that! I think I can just change the check to be `post.scoreChange === 0 ? 0 : 1` and I can avoid having to check `showNegativeKarma`, is that right?

Reply:
Yeah, I guess the setting has already been checked on the backend so we shouldn't have any negative-karma notifications here if it's false so that should be fine 👍